src/txn/txn_db.rs

```diff
@@ -0,0 +1,729 @@
+// Copyright 2021 Yiyuan Liu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::{
+    collections::BTreeMap,
+    ffi::CString,
+    fs, iter,
+    marker::PhantomData,
+    path::{Path, PathBuf},
+    ptr,
+    sync::{Arc, RwLock},
+};
+
+use crate::{
+    column_family::UnboundColumnFamily, db::DBAccess, db_options::OptionsMustOutliveDB, ffi,
+    ffi_util::to_cpath, AsColumnFamilyRef, BoundColumnFamily, ColumnFamilyDescriptor,
+    DBIteratorWithThreadMode, DBRawIteratorWithThreadMode, Direction, Error, IteratorMode, Options,
+    ReadOptions, SnapshotWithThreadMode, Txn, TxnDBOptions, TxnOptions, WriteBatch, WriteOptions,
+    DB, DEFAULT_COLUMN_FAMILY_NAME,
+};
+use libc::{c_char, c_int, size_t};
+
+/// RocksDB TransactionDB.
+///
+/// Please read the official [guide](https://github.com/facebook/rocksdb/wiki/Transactions)
+/// to learn more about RocksDB TransactionDB.
+pub struct TxnDB {
```

Comment:
I'm not strongly opposed to the abbreviated `TxnDB`, but if possible, i like `TransactionDB` (not `Transactiondb` nor `TransactionDb`), especially if the upstream library isn't using `txndb`.

@aleksuss what do you think about this?

Reply:
@ryoqun 100% agree. `TransactionDB` would be better.

Reply:
Ok, I will rename `Txn` to `Transaction`, and `TxnDB` to `TransactionDB`.